lib/src/mamac_device_commands.dart

```diff
@@ -72,3 +77,79 @@ class EditDevice extends SimpleNode {
     (parent as MamacDeviceNode).update(params);
   }
 }
+
+class GetLogsNode extends SimpleNode {
+  static const String isType = 'getLogsNode';
+  static const String pathName = 'Get_Logs';
+  static const String childNodePrefix = 'Logs_';
+
+  static Map<String, dynamic> definition() => {
+        r'$is': isType,
+        r'$name': 'Get Logs',
+        r'$invokable': 'write',
+        r'$params': [],
+        r'$columns': [
+          {'name': 'success', 'type': 'bool', 'default': false},
+          {'name': 'message', 'type': 'string', 'default': ''}
+        ]
+      };
+
+  MamacDevice _device;
+  GetLogsService _getLogsService = new GetLogsService();
+  LinkProvider _link;
+
+  void set device(MamacDevice device) {
```

Comment:
You can do it in the GetHistoryNode `onCreated` method (add it), from within there call something like 
`var device = (parent as MamacDeviceNode).device;`. So that way the GetHistoryNode is responsible for its own configuration and not dependant on being set by the parent node.

The only issue is that the parent node may not be available when the child is created or vice-versa. (I'm not sure if we guarantee order of creation).

Another alternative is to pass the same parameters the MamacDeviceNode gets for its definition and add them to the definition of the GetHistoryNode, and within the `onCreated` you do the same process and create its own device and then change the configuration in the `onCreated` method.